### PR TITLE
Change data capture doc updates for Cloud 2.0

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-9733

Updates all of the docs under `Stream Data` in the docs nav for Cloud 2.0. This PR also touches changefeed SQL page.

### CODE REVIEW

File: src/current/cockroachcloud/stream-changefeed-to-snowflake-aws.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ If you have not done so already, [create a cluster]({% link cockroachcloud/creat
 
 ## Step 2. Connect to your cluster
 
-Refer to [Connect to a CockroachDB Dedicated cluster](https://cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster) for detailed instructions on how to to:
+Refer to [Connect to your cluster](https://cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster) for detailed instructions on how to to:
```
Comment (florence-crl) [inline src/current/cockroachcloud/stream-changefeed-to-snowflake-aws.md:37]: This link will go to connecting to a [Standard Cluster](https://cloud-2-0--cockroachdb-docs.netlify.app/docs/cockroachcloud/connect-to-your-cluster). Is this ok?

Comment (florence-crl) [reply src/current/cockroachcloud/stream-changefeed-to-snowflake-aws.md:37]: I also just noticed that there are 2 Step 2's on this page: Connect and Configure

File: src/current/v23.1/change-data-capture-overview.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ CockroachDB has two implementations of changefeeds:
       <b>Product availability</b>
     </td>
     <td>All products</td>
-    <td>CockroachDB {{ site.data.products.dedicated }}, CockroachDB {{ site.data.products.serverless }}, or with an <a href="enterprise-licensing.html">{{ site.data.products.enterprise }} license</a> in CockroachDB {{ site.data.products.core }}.</td>
+    <td>CockroachDB {{ site.data.products.standard }}, {{ site.data.products.advanced }}, and {{ site.data.products.basic }}, {{ site.data.products.cloud }} clusters, or with an <a href="enterprise-licensing.html">{{ site.data.products.enterprise }} license</a> in CockroachDB {{ site.data.products.core }}.</td>
```
Comment (florence-crl) [inline src/current/v23.1/change-data-capture-overview.md:46]:
```suggestion
    <td>CockroachDB {{ site.data.products.cloud }} clusters ( {{ site.data.products.standard }}, {{ site.data.products.advanced }}, and {{ site.data.products.basic }}) or with an <a href="enterprise-licensing.html">{{ site.data.products.enterprise }} license</a> in CockroachDB {{ site.data.products.core }}.</td>
```

File: src/current/v23.2/change-data-capture-overview.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ CockroachDB has two implementations of changefeeds:
       <b>Product availability</b>
     </td>
     <td>All products</td>
-    <td>CockroachDB {{ site.data.products.dedicated }}, CockroachDB {{ site.data.products.serverless }}, or with an <a href="enterprise-licensing.html">{{ site.data.products.enterprise }} license</a> in CockroachDB {{ site.data.products.core }}.</td>
+    <td>CockroachDB {{ site.data.products.standard }}, {{ site.data.products.advanced }}, and {{ site.data.products.basic }}, {{ site.data.products.cloud }} clusters, or with an <a href="enterprise-licensing.html">{{ site.data.products.enterprise }} license</a> in CockroachDB {{ site.data.products.core }}.</td>
```
Comment (florence-crl) [inline src/current/v23.2/change-data-capture-overview.md:46]:
```suggestion
    <td>CockroachDB {{ site.data.products.cloud }} clusters ({{ site.data.products.standard }}, {{ site.data.products.advanced }}, and {{ site.data.products.basic }}) or with an <a href="enterprise-licensing.html">{{ site.data.products.enterprise }} license</a> in CockroachDB {{ site.data.products.core }}.</td>
```

**Review by florence-crl — APPROVED** [review]
LGTM pending question and suggestions

File: src/current/cockroachcloud/stream-changefeed-to-snowflake-aws.md (hunk 1 above)
Comment (kathancox) [reply src/current/cockroachcloud/stream-changefeed-to-snowflake-aws.md]: That is fine. Not sure what happened with the steps or when. Thanks for catching!

**Pull request merged**